app/[locale]/(authenticated)/profile/ProfileForm.tsx

```diff
@@ -0,0 +1,36 @@
+"use client";
+
+import Form, { FormState } from "@/app/components/Form";
+import { ReactNode, useRef } from "react";
+import updateProfile from "./actions";
+
+export default function ProfileForm({ children }: { children: ReactNode }) {
+  "use client";
+  const formContainerRef = useRef<HTMLDivElement>(null);
+  return (
+    <div ref={formContainerRef}>
+      <Form
+        action={async (
+          _prevState: FormState,
+          data: FormData
+        ): Promise<FormState> => {
+          const formSubmissionState = await updateProfile(_prevState, data);
+          if (formSubmissionState.state === "error") {
```

Comment:
I think the more nextjs way of doing this would be to call `revalidatePath` with the path of the profile view. That will trigger nextjs to rerender the view, which will reset the password field. Then we don't need a separate component to handle this on the client side

Reply:
@arrocke I'm having difficulty getting this to work, could you review my changes and see what I am doing wrong? I addressed the other issues.